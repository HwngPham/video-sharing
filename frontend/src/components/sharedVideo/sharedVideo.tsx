import { useEffect, useState } from "react";
import { Video } from "../../store/constants";
import { getVideoDetail } from "../../services";

export interface SharedVideoProps {
  video: Video;
}
export const SharedVideo = ({ video }: SharedVideoProps) => {
  const [state, setState] = useState<Video>(video);

  useEffect(() => {
    const syncVideoDetail = async () => {
      const detail = await getVideoDetail(video.src);
      setState(detail);
    };

    !video?.version && syncVideoDetail();
  }, [video]);

  return (
    <div className="max-w-sm w-full lg:max-w-full lg:flex border border-blue-200 lg:border-0 lg:mb-4 bg-blue-100 mx-auto lg:mx-4">
      <div
        className="h-48 w-full lg:w-80 lg:h-full flex-none text-center overflow-hidden"
        dangerouslySetInnerHTML={{ __html: state.html ?? "" }}
      ></div>
      <div className="p-4 flex flex-col justify-between leading-normal">
        <div className="mb-8">
          <div className="text-gray-900 font-bold text-xl mb-2">
            {state.title}
          </div>
          <p className="text-gray-700 text-base">{state.author_name}</p>
        </div>
        <div className="flex items-center">
          <div className="text-sm">
            {/* @ts-ignore */}
            <p className="text-gray-900 leading-none">
              shared by {state.user_id}
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};
