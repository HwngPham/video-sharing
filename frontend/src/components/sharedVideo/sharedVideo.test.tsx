import { render } from "@testing-library/react";
import { SharedVideo } from "./sharedVideo";

describe("<SharedVideo />", () => {
  it("matches snapshot", () => {
    const container = render(
      <SharedVideo
        video={{
          id: 1,
          createdAt: new Date(),
          updatedAt: new Date(),
          title: "title",
          desc: "desc",
          html_text: "<p>html_text</p>",
          src: "https://foobar",
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
