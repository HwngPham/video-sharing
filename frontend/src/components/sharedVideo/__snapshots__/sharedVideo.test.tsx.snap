// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SharedVideo /> matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-sm w-full lg:max-w-full lg:flex border border-blue-200 lg:border-0 lg:mb-4 bg-blue-100 mx-auto lg:mx-4"
      >
        <div
          class="h-48 w-full lg:w-80 lg:h-full flex-none text-center overflow-hidden"
        >
          <p>
            html_text
          </p>
        </div>
        <div
          class="p-4 flex flex-col justify-between leading-normal"
        >
          <div
            class="mb-8"
          >
            <div
              class="text-gray-900 font-bold text-xl mb-2"
            >
              title
            </div>
            <p
              class="text-gray-700 text-base"
            >
              desc
            </p>
          </div>
          <div
            class="flex items-center"
          >
            <div
              class="text-sm"
            >
              <p
                class="text-gray-600 leading-none"
              >
                Shared by 
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-sm w-full lg:max-w-full lg:flex border border-blue-200 lg:border-0 lg:mb-4 bg-blue-100 mx-auto lg:mx-4"
    >
      <div
        class="h-48 w-full lg:w-80 lg:h-full flex-none text-center overflow-hidden"
      >
        <p>
          html_text
        </p>
      </div>
      <div
        class="p-4 flex flex-col justify-between leading-normal"
      >
        <div
          class="mb-8"
        >
          <div
            class="text-gray-900 font-bold text-xl mb-2"
          >
            title
          </div>
          <p
            class="text-gray-700 text-base"
          >
            desc
          </p>
        </div>
        <div
          class="flex items-center"
        >
          <div
            class="text-sm"
          >
            <p
              class="text-gray-600 leading-none"
            >
              Shared by 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
